### ISSUE TITLE
Near Live Viewer: Load Near Live Data into new Mantid Imaging Live Viewer Window

### ISSUE THREAD

**Issue by JackEAllen** [mantidproject/mantidimaging#1834]
## Description
**As a:** IMAT user
**I want to:** view live data sent from IMAT cameras to the instrument block house PC during an experiment
**For:** a given dataset being collected during an experiment
**So that:** I can quickly stop a run is the quality of recorded data is poor or the has become misaligned on the rotation stage
**Unlike:** the current solution with is to view data only at the end of an experiment which could result in the need to re-run the experiment which would cause beam time loss for a user with limited time.
**Our solution is:** to either poll or use notifications to watch for updates in user selected directory for new images being written to the directory and loading them into a new window within Mantid Imaging so the user can quickly evaluate quality of data in a familiar UI.

### Workflow
A live view would run on the cabin PC within Mantid Imaging. User selects a folder, software polls/notification of change in folder for new images and displays the latest in Mantid Imaging Live Viewer window. Would be nice to have normal Mantid Imaging stack viewing controls (colour levels, stack scrolling)

## Acceptance criteria 
- [ ] User can select a folder 
- [ ] Most recent image is displayed 
- [ ] GUI updates as new images arrive 

## Additional Potential Features
- [ ] Image stack interaction 
- [ ] Handling of image type (sample, flat, dark, 180) 
- [ ] Pausing or smart updates 
- [ ] On demand loading from disk, to avoid holding everything in RAM 
- [ ] File pruning – may be a good place for removing bad images 
- [ ] Show useful metadata 

## Enabling Conditions
- Investigation of Dask for loading from disk larger than RAM Data where total data size is unknown.
- Polling vs Notification (inotify, Qt)

## Blocking Conditions
__*What needs to be in place before this issue can be closed?*__

### Business Outcome Hypothesis
The current manual process risks having checks missed, so experiment setup issues may not be noticed, and many hours of beam time can be lost. This has happened a few times in the past. If a user only has 2 days of beam time. This can drastically reduce the amount of time available for that user to perform their experiment as up to half a day can be lost. 

A simple to use live viewer would mean that issues were spotted and corrected early, before significant time is lost. 

Setting up the live viewer would take less than a minute and require no further user interaction to continue seeing live updates. Saving around half an hour on a typical experiment. 
 
Improved user experience, more professional. Upholds ISIS’s reputation. Users are more likely to come back. 

### Leading Indicators
- Can detect new image arrivals
- Mantid Imaging can display the most recent image
- The number of manual steps are reduced

### In Scope
Directory selection 
New TIFF file detection 
User interface to display images 
Interaction with stack of images (image slider, zoom, brightness/contrast levels, pixel values) 

### Out of Scope
- Data streaming (this work will help with doing this in the future) 
- Non-tiff file formats, event mode data  
- Complex operations and reconstruction on live data

### Non-functional Requirements (NFRs)
- RAM exhaustion. 
- Needs to run on cabin computers (Windows), without effecting other operations. 
- Near real time – display image within 10 seconds of arriving on cabin PC (there may be other delays between recording and arriving on the PC) 
- Correct handling of partial files, corrupt files 
- Satisfied with Polling or notification speed and recourses usage to detect new images added to directory and that files aren't missed without using too many recourses.
- Works with outliers in the data
